src/accountsdb/db.zig

```diff
@@ -4052,19 +4061,65 @@ pub const BenchmarkAccountsDBSnapshotLoad = struct {
             snapshot.accounts_db_fields,
             bench_args.n_threads,
             allocator,
-            1_500,
+            250, // should be safe for testnet snapshots
         );
 
-        // sanity check
-        const accounts_hash, const total_lamports = try accounts_db.computeAccountHashesAndLamports(.{
-            .FullAccountHash = .{
-                .max_slot = accounts_db.largest_rooted_slot.load(.monotonic),
-            },
-        });
-        std.debug.print("r: hash: {}, lamports: {}\n", .{ accounts_hash, total_lamports });
-
         return duration.asNanos();
     }
+
+    pub fn verifySnapshot(bench_args: BenchArgs) !u64 {
+        const allocator = std.heap.c_allocator;
+        const logger = Logger.init(allocator, .debug);
+        defer logger.deinit();
+        logger.spawn();
+
+        const snapshot_dir = std.fs.cwd().openDir(dir_path, .{ .iterate = true }) catch {
+            std.debug.print("need to setup a snapshot in {s} for this benchmark...\n", .{dir_path});
+            return 0;
+        };
+        const snapshot_files = try SnapshotFiles.find(allocator, snapshot_dir);
+
+        std.fs.cwd().access(accounts_path, .{}) catch {
+            const archive_file = try snapshot_dir.openFile(snapshot_files.full_snapshot.snapshotNameStr().constSlice(), .{});
+            defer archive_file.close();
+            try parallelUnpackZstdTarBall(
+                allocator,
+                logger,
+                archive_file,
+                snapshot_dir,
+                bench_args.n_threads,
+                true,
+            );
+        };
+
+        var snapshots = try AllSnapshotFields.fromFiles(
+            allocator,
+            logger,
+            snapshot_dir,
+            snapshot_files,
+        );
+        defer snapshots.deinit(allocator);
+        const snapshot = try snapshots.collapse();
+
+        var accounts_db = try AccountsDB.init(allocator, logger, snapshot_dir, .{
+            .number_of_index_bins = 32,
+            .use_disk_index = bench_args.use_disk,
+        }, null);
+        defer accounts_db.deinit();
+
+        var accounts_dir = try std.fs.cwd().openDir(accounts_path, .{ .iterate = true });
+        defer accounts_dir.close();
+
+        const full_snapshot = snapshots.full;
+        const validate_duration = try accounts_db.validateLoadFromSnapshot(
```

Comment:
this is going to fail because accounts-db hasnt been loaded, so youre going to validate nothing 

probably want `accounts_db.validateLoadFromSnapshot(` right after `loadSnapshot` benchmark and print two values 'LOAD: {duration}', 'VALIDATE: {duration}' because both operations are expensive (dont want to have to re-load again just to validate)

-- see how we print the read/writes in `BenchmarkAccountsDB`

Reply:
makes sense. i can print both, but i will return the time it took to validate from the benchmark. need that for the benchmark UI, since we already have a benchmark that times how long it takes to load.

Reply:
fixed in a4dcc7e0b044b79047054fe41e6ed1d0293afaa9, now call `loadSnapshot` then `validateLoadFromSnapshot`.